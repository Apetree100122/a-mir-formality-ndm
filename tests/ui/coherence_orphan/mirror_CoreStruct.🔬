[
    crate core {
        trait CoreTrait<> where [] {}
        struct CoreStruct<> where [] {}

        trait Mirror<> where [] {
            type Assoc<> : [] where [];
        }

        impl<ty T> Mirror<> for T where [] {
            type Assoc<> = T where [];
        }
    },
    crate foo {
        impl<> CoreTrait<> for <CoreStruct<> as Mirror<>>::Assoc<> where [] {}
    }
]
