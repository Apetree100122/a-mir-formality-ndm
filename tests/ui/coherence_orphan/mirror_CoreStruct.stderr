Error: orphan_check(impl CoreTrait for <CoreStruct as Mirror>::Assoc where [] { })

Caused by:
    failed to prove {@ IsLocal(CoreTrait(<CoreStruct as Mirror>::Assoc))} given {}, got {}
